daemon/keys.go

```diff
@@ -0,0 +1,43 @@
+// +build linux
+
+package daemon
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	rootKeyFile  = "/proc/sys/kernel/keys/root_maxkeys"
+	rootKeyLimit = 1000000
+)
+
+// ModifyRootKeyLimit checks to see if the root key limit is set to
+// atleast 1000000.
+func ModifyRootKeyLimit() error {
+	value, err := readRootKeyLimit()
+	if err != nil {
+		return err
+	}
+	if value < rootKeyLimit {
```

Comment:
@crosbymichael I don't like this, because we're changing global distribution policy. There's almost certainly some side-effects -- I'd prefer if we made the keyring joining optional (as you mentioned in opencontainers/runc#818).